error.go

```diff
@@ -38,6 +38,15 @@ type Error interface {
 
 var _ Error = proto.RedisError("")
 
+func isContextError(err error) bool {
+	switch err {
+	case context.Canceled, context.DeadlineExceeded:
```

Comment:
I'm not familiar with the code to know how the errors get passed here, but this approach would only match directly `context.Canceled` and `context.DeadlineExceeded`, it won't match error wrapping those errors.

In general (https://pkg.go.dev/errors#section-documentation) we should use `errors.Is(err, context.Canceled)` over `err == context.Canceled`, unless there's a reason we only want to match direct exact errors.

(not sure whether this is the reason this PR does not fix the issue we saw in https://github.com/redis/go-redis/issues/3282#issuecomment-2672158309)

Reply:
Generally errors are not wrapped in the `go-redis` package and this error should occur within the package. I do agree that with the new version there are tons of things that can be improved, error handling included. Currently we are focused on improving the stability, testing infrastructure , bug fixes and redis 8 support. Once those are addressed we will move to other improvements that are nice to have... I will continue looking into your issue and continue the discussion in its thread.